### PR TITLE
HPCC-15244 Add support for Nagios XI

### PR DESCRIPTION
**Pull request by garonsky**
- Improve CLI
- Support for Nagios XI

Signed-off-by: Gleb Aronsky <gleb.aronsky@lexisnexis.com>

### CODE REVIEW

**Comment by garonsky** [comment]
@Michael-Gardner please review cmake changes
@wangkx please review

File: hpcc-nagios-tools/HPCCNagiosToolSet.cpp [hunk 1]
```diff
@@ -431,17 +442,39 @@ bool CHPCCNagiosToolSet::generateHostGroupsConfigurationFile(const char* pOutput
...
+
+                if (len == 0)
+                    strHostConfig.append(P_NAGIOS_HOSTS_GROUP_END_BRACKET);
+                else
+                    strHostConfig.append(", ");
+            }
+        }
+        else
+        {
+            appendHeader(strHostConfig);
```
Comment (wangkx) [inline hpcc-nagios-tools/HPCCNagiosToolSet.cpp]: Question: if (m_bGenerateCatchAllHostGroup), you clean  the strHostConfig before appendHeader(), but not here. Is it intentional?

File: hpcc-nagios-tools/main.cpp [hunk 2]
```diff
@@ -765,6 +839,16 @@ int main(int argc, char *argv[])
             i++;
             strURL.set(argv[i]);
         }
+        else if(stricmp(argv[i], "-set_catch_all_hostgroup") == 0)
+        {
+            CHPCCNagiosToolSet::m_bGenerateCatchAllHostGroup = true;
+            CHPCCNagiosToolSet::m_CatchAllHostGroupName.set(argv[++i]);
```
Comment (wangkx) [inline hpcc-nagios-tools/main.cpp]: Do you want to check if (argv[i] == nullptr || *argv[i] == 0) here also?

**Comment by richardkchapman** [comment]
@garonsky Back to you

**Comment by garonsky** [comment]
@Michael-Gardner the dependency files were removed because this package does not have a binary dependency on them. 
@wangkx @Michael-Gardner changes made based on feedback.  Commits squashed. Please review again.

**Comment by wangkx** [comment]
Looks fine for me.

**Pull request merged**